src/App.js

```diff
@@ -1,16 +1,37 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
 
 const App = () => {
+
+  const [entriesData, setEntriesData] = useState(chatMessages);
+
+  const updateLike = (entryId) => {
+    const updatedEntries = entriesData.map(entry => {
+      if (entry.id === entryId.id) {
+        return {...entry, liked: !entry.liked};
+      } else {
+        return entry;
+      }
+    });
+    setEntriesData(updatedEntries)
+  }
+
+  const countLikes = entriesData.filter((entry) => entry.liked).length;
+
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>Chat between Vladimir and Estragon</h1>
+        <p>{countLikes} ❤️s</p>
       </header>
       <main>
-        {/* Wave 01: Render one ChatEntry component
-        Wave 02: Render ChatLog component */}
+        <ChatLog 
+          entries={chatMessages}
```

Comment:
Note that you're passing down the initially imported chat data rather than the `entriesData` controlled by state. This is a problem because all of the displayed chat data will not correctly reflect any changes to the likes (inspecting the elements in the browser will reveal `liked` will always be `false` even when it has been liked).

Your app still works because you control whether a red or white heart should be displayed entirely at the `chatEntry` level, liking a message does correctly update the `entriesData` in state, and the calculated likes logic also correctly uses the `entriesData` in state. So, your logic gets around this problem of having the wrong chat data, but it's best to pass down the chat data from state so that the props for each entry correctly reflect the current state of the entry. Having accurate info in the props will also mean you won't need another state variable for the heart button.